src/main/java/com/commercetools/sync/services/CustomObjectService.java

```diff
@@ -0,0 +1,74 @@
+package com.commercetools.sync.services;
+
+
+import com.commercetools.sync.commons.models.NonResolvedReferencesCustomObject;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+public interface CustomObjectService {
```

Comment:
I would name it e.g. `LazyResolutionService`. The service itself is about keeping track of `productDrafts` with references that cannot be resolved yet. So the service can be used to lazily create/update those drafts as soon as the references can be resolved.

So the fact that the inner implementation uses CTP `CustomObject`s is an implementation specific/detail. It is not the main focus of the service.

Reply:
yes, It makes sense :)

Reply:
63ff515cdd97b4350b6cc8d1b9fee25c6b532345